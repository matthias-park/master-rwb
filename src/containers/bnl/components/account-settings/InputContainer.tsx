import React from 'react';
import clsx from 'clsx';
import Form from 'react-bootstrap/Form';
import LoadingButton from '../../../../components/LoadingButton';
import { useForm, FormProvider } from 'react-hook-form';
import TextInput from '../../../../components/customFormInputs/TextInput';
import { useI18n } from '../../../../hooks/useI18n';
interface Props {
  title: string;
  defaultValue?: number | string;
  buttonText: string | JSX.Element;
  buttonClassName?: string;
  loading?: boolean;
  min?: number | string;
  max?: number | string | null;
  onSubmit: (inputValue: number) => Promise<boolean | void>;
  quickAmounts?: number[];
  disabled?: boolean;
  currency?: string;
  header?: JSX.Element;
  subText?: string;
  inputTitle?: string;
  validationErrorPrefix: string;
  amountButtonsUsed?: (used: boolean) => void;
}

const InputContainer = ({
  title,
  defaultValue = '',
  validationErrorPrefix,
  buttonText,
  buttonClassName,
  loading,
  onSubmit,
  quickAmounts = [],
  min = 0,
  max,
  disabled,
  currency = '',
  subText,
  header,
  inputTitle,
  amountButtonsUsed,
}: Props) => {
  const { t } = useI18n();
  const formMethods = useForm<{
    amount: string;
  }>({
    mode: 'onBlur',
    defaultValues: {
      amount: defaultValue.toString(),
    },
  });

  const { formState, watch, setValue, reset } = formMethods;

  const resetAmount = () =>
    reset(
      {
        amount: defaultValue.toString(),
      },
      {
        keepErrors: false,
        keepDefaultValues: true,
        keepDirty: false,
        keepIsSubmitted: false,
        keepIsValid: false,
        keepSubmitCount: false,
        keepTouched: false,
        keepValues: false,
      },
    );

  React.useEffect(() => {
    if (formState.isSubmitSuccessful) {
      resetAmount();
    }
  }, [formState, reset]);

  const handleSubmit = () => {
    formMethods.handleSubmit(async ({ amount }) => onSubmit(Number(amount)))();
  };
  const validateAmount = (value: string) => {
    if (!value) return true;
    let inputAmount = Number(value);
    const minNumber = Number(min);
    const maxNumber = Number(max);
    if (min != null && !isNaN(minNumber) && inputAmount < minNumber) {
      return t(`${validationErrorPrefix}amount_below_minimum`);
    } else if (max != null && !isNaN(maxNumber) && inputAmount > maxNumber) {
      return t(`${validationErrorPrefix}amount_above_maximum`);
    }
    return true;
  };
  return (
    <div className="input-container mb-4">
      {header && header}
      <p data-testid="title" className="input-container__title text-14 mb-2">
        {title}
      </p>
      <FormProvider {...formMethods}>
        <Form.Group className="w-100">
          <div className="quick-amounts">
            {quickAmounts.map(value => (
              <button
                key={value}
                className={clsx(
                  'quick-amounts__btn',
                  watch('amount') === value.toString() && 'active',
                )}
                onClick={() => {
                  setValue('amount', value.toString(), {
                    shouldValidate: true,
                    shouldDirty: true,
                  });
                  amountButtonsUsed?.(true);
                }}
                disabled={disabled}
              >
                {`${currency} ${value}`}
              </button>
            ))}
          </div>
          <TextInput
            id="amount"
            className="input-container__input"
            title={inputTitle}
            maskedInput={{
              allowEmptyFormatting: true,
              prefix: `${currency} `,
              thousandSeparator: true,
              allowNegative: false,
            }}
            rules={{
              validate: validateAmount,
            }}
            onBlur={() => {
              const amount = watch('amount', '');
              if (!amount.length) {
                resetAmount();
              }
            }}
            onChange={() => {
              amountButtonsUsed?.(false);
            }}
            disabled={disabled}
            onEnterPress={() => handleSubmit()}
            clearDefaultValueOnFocus
            defaultValue={defaultValue.toString()}
          />
        </Form.Group>
        {subText && <small className="mb-2">{subText}</small>}
        <LoadingButton
          variant="primary"
          disabled={
            !formState.isDirty ||
            !watch('amount', '') ||
            validateAmount(watch('amount', '')) !== true
          }
          className={clsx(buttonClassName ? buttonClassName : '')}
          onClick={handleSubmit}
          data-testid="button"
          loading={!!formState.isSubmitting || loading}
        >
          {buttonText}
        </LoadingButton>
      </FormProvider>
    </div>
  );
};

export default InputContainer;
