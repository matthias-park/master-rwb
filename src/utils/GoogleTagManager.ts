import {
  IDataLayer,
  ISnippets,
  ISendToGTM,
  ISetupGTM,
  ISnippetsParams,
} from '../types/GoogleTagManager';

const getDataLayerSnippet = (
  dataLayer: Pick<IDataLayer, 'dataLayer'>['dataLayer'],
  dataLayerName: Pick<
    IDataLayer,
    'dataLayerName'
  >['dataLayerName'] = 'dataLayer',
): Pick<ISnippets, 'gtmDataLayer'>['gtmDataLayer'] =>
  `window.${dataLayerName} = window.${dataLayerName} || []; window.${dataLayerName}.push(${JSON.stringify(
    dataLayer,
  )})`;

const getIframeSnippet = (
  id: Pick<ISnippetsParams, 'id'>['id'],
  environment?: Pick<ISnippetsParams, 'environment'>['environment'],
) => {
  let params = ``;
  if (environment) {
    const { gtm_auth, gtm_preview } = environment;
    params = `&gtm_auth=${gtm_auth}&gtm_preview=${gtm_preview}&gtm_cookies_win=x`;
  }
  return `<iframe src="https://www.googletagmanager.com/ns.html?id=${id}${params}" height="0" width="0" style="display:none;visibility:hidden" id="tag-manager"></iframe>`;
};

const getGTMScript = (
  dataLayerName: Pick<ISnippetsParams, 'dataLayerName'>['dataLayerName'],
  id: Pick<ISnippetsParams, 'id'>['id'],
  environment?: Pick<ISnippetsParams, 'environment'>['environment'],
) => {
  let params = ``;
  if (environment) {
    const { gtm_auth, gtm_preview } = environment;
    params = `+"&gtm_auth=${gtm_auth}&gtm_preview=${gtm_preview}&gtm_cookies_win=x"`;
  }
  return `
      (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
        'https://www.googletagmanager.com/gtm.js?id='+i+dl${params};f.parentNode.insertBefore(j,f);
      })(window,document,'script','${dataLayerName}','${id}');
    `;
};
/**
 * Function to setup the Google Tag Manager
 * @param params - The snippets params
 */
const setupGTM = (params: ISnippetsParams): ISetupGTM => {
  const getDataLayerScript = (): HTMLElement => {
    const dataLayerScript = document.createElement('script');
    dataLayerScript.innerHTML = getDataLayerSnippet(
      params.dataLayer,
      params.dataLayerName,
    );
    return dataLayerScript;
  };

  const getNoScript = (): HTMLElement => {
    const noScript = document.createElement('noscript');
    noScript.innerHTML = getIframeSnippet(params.id, params.environment);
    return noScript;
  };

  const getScript = (): HTMLElement => {
    const script = document.createElement('script');
    script.innerHTML = getGTMScript(
      params.dataLayerName,
      params.id,
      params.environment,
    );
    return script;
  };

  return {
    getDataLayerScript,
    getNoScript,
    getScript,
  };
};

/**
 * Function to init the GTM
 * @param dataLayer - The dataLayer
 * @param dataLayerName - The dataLayer name
 * @param environment - Specify the custom environment to use
 * @param id - The ID of the GTM
 */
export const initGTM = ({
  dataLayer,
  dataLayerName,
  environment,
  id,
}: ISnippetsParams): void => {
  const gtm = setupGTM({
    dataLayer,
    dataLayerName,
    environment,
    id,
  });

  const dataLayerScript = gtm.getDataLayerScript();
  const script = gtm.getScript();
  const noScript = gtm.getNoScript();

  document.head.insertBefore(script, document.head.childNodes[0]);
  document.head.insertBefore(dataLayerScript, document.head.childNodes[0]);
  document.body.insertBefore(noScript, document.body.childNodes[0]);
};

/**
 * Function to send the events to the GTM
 * @param dataLayerName - The dataLayer name
 * @param data - The data to push
 */
export const sendToGTM = ({ dataLayerName, data }: ISendToGTM): void =>
  window[dataLayerName].push(data);
