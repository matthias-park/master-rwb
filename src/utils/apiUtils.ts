import fetch from 'isomorphic-unfetch';
import { ConfigInterface, mutate } from 'swr';
import { RailsApiResponseFallback } from '../constants';
import RailsApiResponse from '../types/api/RailsApiResponse';
import { RegistrationPostalCodeAutofill } from '../types/api/user/Registration';
import { cleanPostBody } from '.';
import * as Sentry from '@sentry/react';

export const formatUrl = (
  url: string,
  params: { [key: string]: string | number | undefined } = {},
): string => {
  const paramsArr = Object.keys(params)
    .filter(key => params[key])
    .map(key => `${key}=${params[key]}`);
  return `${url}${paramsArr.length ? '?' : ''}${paramsArr.join('&')}`;
};

interface GetApiOptions {
  responseText?: boolean;
  cache?: RequestCache;
}

export const getApi = <T>(url: string, options?: GetApiOptions): Promise<T> => {
  const config: RequestInit = {
    mode: 'cors',
    credentials: 'include',
    cache: options?.cache,
    headers: new Headers(),
  };
  const getUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(getUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    if (options?.responseText) {
      return res.text();
    }
    return res.json();
  });
};

export interface PostOptions {
  formData?: boolean;
}

export const postApi = <T>(
  url: string,
  body?: {
    [key: string]: number | string | Blob | boolean | undefined | null | object;
  },
  options: PostOptions = {},
): Promise<T> => {
  const config: RequestInit = {
    method: 'post',
    mode: 'cors',
    credentials: 'include',
    headers: new Headers(),
  };
  if (body) {
    if (options.formData) {
      const formData = new FormData();
      const formBody = cleanPostBody(body);
      for (const key in formBody) {
        if (Array.isArray(formBody[key])) {
          for (const item of formBody[key]) {
            formData.append(key, item as string | Blob);
          }
        } else {
          formData.append(key, formBody[key] as string | Blob);
        }
      }
      config.body = formData;
    } else {
      config.body = JSON.stringify(cleanPostBody(body));
      (config.headers as Headers).append('Content-Type', 'application/json');
    }
  }
  const postUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(postUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    return res.json();
  });
};

export const SwrFetcherConfig: ConfigInterface<
  any,
  any,
  (...args: any) => any
> = {
  fetcher: getApi,
  revalidateOnFocus: false,
  revalidateOnReconnect: false,
  errorRetryCount: 2,
  dedupingInterval: 30000,
  focusThrottleInterval: 10000,
  errorRetryInterval: 10000,
  onError: (err, key) => {
    if (![401, 403].includes(err.status)) {
      Sentry.captureMessage(
        `Request failed ${key} with status ${err.status}`,
        Sentry.Severity.Fatal,
      );
    }
  },
};

export const formatSuccesfullRailsApiResponse = <T>(
  data: T,
): RailsApiResponse<T> => ({
  Code: 0,
  Data: data,
  Message: '',
  Success: true,
});

export const API_VALIDATIONS = {
  email: async (email: string): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/email',
      {
        email,
      },
    ).catch(err => err);
    return res;
  },
  personalCode: async (
    personal_code: string,
  ): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/personal_code',
      {
        personal_code,
      },
    ).catch((err: RailsApiResponse<{}>) => err);
    return res;
  },
  postalCode: async (
    post_code: string,
  ): Promise<RailsApiResponse<RegistrationPostalCodeAutofill | null>> => {
    const res = await postApi<
      RailsApiResponse<RegistrationPostalCodeAutofill | null>
    >('/restapi/v1/registration/check/post_code', {
      post_code,
    }).catch((err: RailsApiResponse<null>) => err);
    return res;
  },
};
