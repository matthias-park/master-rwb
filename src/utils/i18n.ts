import { ConfigRoute } from '../types/Config';
type Symbols = { [key: string]: { [key: string]: string } };

const i18n = () => {
  let locale = '';
  const symbols: Symbols = {};

  return {
    set(lang: string, data: unknown = {}) {
      symbols[lang] = Object.assign(symbols[lang] || {}, data);
    },

    locale(lang?: string) {
      return (locale = lang || locale);
    },

    table(lang: string) {
      return symbols[lang];
    },

    t(key: string, lang?: string) {
      const val = symbols?.[lang || locale]?.[key] || '';
      return val;
    },
  };
};

export const getRedirectLocalePathname = (
  availableLocales: string[],
  defaultLocale: string,
  availableRoutes: ConfigRoute[],
) => {
  const windowPaths = window.location.pathname.split('/');
  let urlLocale = defaultLocale;
  let urlPaths = window.location.pathname;

  if (availableLocales.includes(windowPaths[1])) {
    urlLocale = windowPaths[1];
  } else if (availableRoutes.some(route => route.path === urlPaths)) {
    urlPaths = `/${urlLocale}${urlPaths}`;
  } else {
    const urlPathWithoutFirstPath = urlPaths.replace(`/${windowPaths[1]}`, '');
    if (availableRoutes.some(route => route.path === urlPathWithoutFirstPath)) {
      urlPaths = `/${urlLocale}${urlPathWithoutFirstPath}`;
    } else {
      urlPaths = `/${urlLocale}`;
    }
  }

  if (window.location.pathname !== urlPaths) {
    window.history.replaceState({}, '', urlPaths);
  }
  return urlLocale;
};

export const setLocalePathname = (newlocale: string) => {
  const paths = window.location.pathname.split('/');
  paths[1] = newlocale;
  const newPath = paths.join('/');
  window.history.pushState({}, '', newPath);
};

export type I18n = ReturnType<typeof i18n>;
export default i18n;
