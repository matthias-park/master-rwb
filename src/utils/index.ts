import { PagesName } from '../constants';
import { PageConfig } from '../types/api/PageConfig';
import Lockr from 'lockr';
import { PostItem } from '../types/api/Posts';
import dayjs from 'dayjs';

export const sortAscending = (a: number, b: number) => a - b;
export const sortDescending = (a: number, b: number) => b - a;

export function throttle(func: Function, limit: number): Function {
  let inThrottle: boolean;

  return function (this: any): any {
    const args = arguments;
    const context = this;
    if (!inThrottle) {
      inThrottle = true;
      func.apply(context, args);
      setTimeout(() => (inThrottle = false), limit);
    }
  };
}

export const stringToMiliseconds = (time: string): number | null => {
  const parsed = time.split(':');
  if (parsed.length === 3) {
    return miliseconds(Number(parsed[0]), Number(parsed[1]), Number(parsed[2]));
  }
  return null;
};
export const miliseconds = (
  hrs: number = 0,
  min: number = 0,
  sec: number = 0,
): number => {
  return (hrs * 60 * 60 + min * 60 + sec) * 1000;
};
export const removeFalsyFromObject = (obj: any) =>
  Object.keys(obj).forEach(
    k => !obj[k] && obj[k] !== undefined && delete obj[k],
  );

export const structuredBankCommunications = (barcode): string => {
  if (!barcode) return '';
  const rExp = /\d+/;
  const moduloOf97 = barcode.match(rExp)[0] % 97;
  return '00' + barcode + moduloOf97.toString().padStart(2, '0');
};

export const formatNavigationRoutes = (constants: PageConfig) => {
  const navigationRoutes = constants.navigation_routes
    ? [...constants.navigation_routes]
    : [];
  if (constants.content_pages) {
    for (const [key, value] of Object.entries(constants.content_pages)) {
      const name = Array.isArray(value) ? value[0] : value;
      const slug = Array.isArray(value) ? value[1] : key;
      const contentPagePath = `${slug.startsWith('/') ? '' : '/'}${slug}`;
      if (!navigationRoutes.some(route => route.path === contentPagePath)) {
        navigationRoutes.push({
          id:
            Array.isArray(value) && PagesName[key]
              ? Number(key)
              : PagesName.TemplatePage,
          path: contentPagePath,
          name,
        });
      }
    }
  }
  return navigationRoutes.sort((a, b) =>
    sortDescending(a.path.length, b.path.length),
  );
};

export const clearUserLocalStorage = () => {
  const keys = ['transactions-date-to', 'transactions-date-from'];
  for (const key of keys) {
    localStorage.removeItem(key);
  }
};

export const cleanPostBody = (obj: { [key: string]: unknown }) =>
  Object.entries(obj).reduce((obj, [key, value]) => {
    if (!key.includes('temp_field_') && value != null) {
      obj[key] = typeof value === 'string' ? value.trim() : value;
    }
    return obj;
  }, {});

export const errorHandler = (event: ErrorEvent) => {
  if (event.error?.stack?.includes('kambi-chunk')) {
    const kambiErrorId = 'kambi-error-reload';
    const kambiErrorRetryCount = Lockr.get(kambiErrorId, 0) + 1;
    if (kambiErrorRetryCount < 3) {
      Lockr.set(kambiErrorId, kambiErrorRetryCount);
      window.location.reload();
    }
  }
};

export const filterPromotionsList = (promotions: PostItem[]): PostItem[] =>
  promotions.filter(promotion => {
    const isNotExpired =
      !promotion.expiration_date ||
      dayjs(promotion.expiration_date).isAfter(dayjs());
    const isPublished =
      promotion.publish_date && dayjs(promotion.publish_date).isBefore(dayjs());
    const isVisible = promotion.visible;
    return isNotExpired && isPublished && isVisible;
  });
