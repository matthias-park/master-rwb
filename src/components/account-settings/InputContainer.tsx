import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { enterKeyPress } from '../../utils/uiUtils';
import Spinner from 'react-bootstrap/Spinner';

interface Props {
  title: string;
  placeholder: string;
  buttonText: string;
  loading?: boolean;
  onSubmit: (inputValue: number) => void;
}

const InputContainer = ({
  title,
  placeholder,
  buttonText,
  loading,
  onSubmit,
}: Props) => {
  const [inputValue, setInputValue] = useState<string>('');
  const handleSubmit = () => onSubmit(Number(inputValue));
  return (
    <div className="input-container mb-4">
      <p className="input-container__title text-14 mb-2">{title}</p>
      <Form.Group className="w-100">
        <Form.Control
          type="number"
          id="amount"
          placeholder={placeholder}
          value={inputValue}
          onKeyUp={e => enterKeyPress(e, handleSubmit)}
          onChange={e => setInputValue(e.target.value)}
          className="input-container__input"
        ></Form.Control>
        <small className="form-group__error-msg">Error message</small>
      </Form.Group>
      <Button variant="primary" disabled={loading} onClick={handleSubmit}>
        {loading && (
          <Spinner
            as="span"
            animation="border"
            size="sm"
            role="status"
            className="mr-1"
          />
        )}
        {buttonText}
      </Button>
    </div>
  );
};

export default InputContainer;
