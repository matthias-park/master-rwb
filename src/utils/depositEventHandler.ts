import { CustomWindowEvents, ComponentName } from '../constants';
import RequestReturn from '../types/api/deposits/RequestReturn';
import RailsApiResponse from '../types/api/RailsApiResponse';
import depositEventHandlerProps from '../types/DepositEventHandler';
import { postApi } from './apiUtils';
import * as Sentry from '@sentry/react';

const depositEventHandler = ({
  eventType,
  eventData,
  setCustomHtml,
  depositStatus,
  setApiError,
  setDepositLoading,
  t,
  depositRequestId,
  bankId,
  locale,
  setCurrentStep,
  disableModal,
  depositForm,
}: depositEventHandlerProps) => {
  switch (eventType) {
    case CustomWindowEvents.DepositPaymentSuccess: {
      setApiError(null);
      setCustomHtml(null);
      depositStatus.startCheckingStatus();
      break;
    }
    case CustomWindowEvents.DepositPaymentError: {
      if (typeof eventData === 'string') {
        Sentry.captureMessage(`deposit error: ${eventData}`);
        setApiError({ message: eventData, variant: 'danger' });
      } else {
        setApiError({ message: t('api_response_failed'), variant: 'danger' });
      }
      setCustomHtml(null);
      setDepositLoading(false);
      break;
    }
    case CustomWindowEvents.DepositPaymentWarning: {
      if (typeof eventData === 'string') {
        setApiError({ message: eventData, variant: 'warning' });
      }
      break;
    }
    case CustomWindowEvents.DepositRequestReturn: {
      if (eventData && typeof eventData === 'object') {
        const data: any = {
          ...eventData,
          depositRequestId,
        };
        if (bankId === 189) {
          data.browserInfo = {
            acceptHeader: '*',
            colorDepth: window.screen.colorDepth,
            javaEnabled: false,
            language: locale,
            screenHeight: window.innerHeight,
            screenWidth: window.innerWidth,
            timeZoneOffset: new Date().getTimezoneOffset(),
            userAgent: window.navigator.userAgent,
          };
        }
        postApi<RailsApiResponse<RequestReturn>>(
          '/restapi/v1/deposits/request_return',
          {
            bank_id: bankId.toString(),
            data: JSON.stringify(data),
          },
        )
          .then(res => {
            if (res.Data.OK) {
              if (res.Data.AdditionalData?.redirectUrl) {
                if (depositForm) {
                  window.open(res.Data.AdditionalData.redirectUrl, '_blank');
                  disableModal?.(ComponentName.QuickDepositModal);
                } else {
                  window.location.href = res.Data.AdditionalData.redirectUrl;
                }
                return;
              }
              if (res.Data.AdditionalData?.verifyPaymentData) {
                return window.postMessage(
                  JSON.stringify({
                    eventType: CustomWindowEvents.DepositVerifyPayment,
                    eventData: res.Data.AdditionalData.verifyPaymentData,
                  }),
                  '*',
                );
              }
              setCustomHtml(null);
              depositStatus.startCheckingStatus();
            } else if (res.Data.Error) {
              setCurrentStep?.(1);
              setApiError({
                message: t('api_response_failed'),
                variant: 'danger',
              });
              Sentry.captureMessage(
                `Got DepositRequestReturn error: ${res.Data.Error}`,
                Sentry.Severity.Info,
              );
            }
          })
          .catch(() => null);
      } else {
        Sentry.captureMessage(
          'Got DepositRequestReturn without data',
          Sentry.Severity.Fatal,
        );
      }
    }
  }
};
export default depositEventHandler;
