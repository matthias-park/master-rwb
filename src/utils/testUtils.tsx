import React, { useEffect, useState } from 'react';
import { render } from '@testing-library/react';
import { SWRConfig } from 'swr';
import { SwrFetcherConfig } from './apiUtils';
import { ToastProvider } from 'react-toast-notifications';
import { HelmetProvider } from 'react-helmet-async';
import { Provider } from 'react-redux';
import createStoreAsync from '../state';

const AllTheProviders = ({ children }) => {
  const [state, setState] = useState<any>(null);
  useEffect(() => {
    createStoreAsync().then(store => setState(store));
  }, []);
  if (!state) return null;
  return (
    <Provider store={state}>
      <SWRConfig value={SwrFetcherConfig}>
        <ToastProvider>
          <HelmetProvider>{children}</HelmetProvider>
        </ToastProvider>
      </SWRConfig>
    </Provider>
  );
};

const customRender = (ui, options: any = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
