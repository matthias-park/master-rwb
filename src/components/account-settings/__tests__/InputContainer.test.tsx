import React from 'react';
import { render, cleanup, fireEvent } from '../../../utils/testUtils';
import renderer from 'react-test-renderer';
import InputContainer from '../InputContainer';
import { act } from 'react-dom/test-utils';

beforeEach(cleanup);

const titleContent = 'titlePlaceholder';
const buttonContent = 'buttonPlaceholder';

test('displays title correctly', async () => {
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
    />,
  );
  expect(getByTestId('title')).toHaveTextContent(titleContent);
});

test('displays button correctly', async () => {
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
    />,
  );
  expect(getByTestId('button')).toHaveTextContent(buttonContent);
});

test("doesn't display spinner correctly", async () => {
  const { queryByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
      loading={false}
    />,
  );
  expect(queryByTestId('spinner')).toBeNull();
});
test('displays spinner correctly', async () => {
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
      loading={true}
    />,
  );
  expect(getByTestId('spinner')).toBeInTheDocument();
});
test('submit value correctly', async () => {
  const newEnteredValue = '500';
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={value => {
        expect(value).toBe(Number(newEnteredValue));
      }}
      loading={false}
    />,
  );
  fireEvent.change(getByTestId('input'), {
    target: { value: newEnteredValue },
  });
  await act(async () => {
    fireEvent.click(getByTestId('button'));
  });
});

test('submit value less than min', async () => {
  const newEnteredValue = '10';
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      min="200"
      onSubmit={value => {
        expect(value).toBe(Number(200));
      }}
      loading={false}
    />,
  );
  fireEvent.change(getByTestId('input'), {
    target: { value: newEnteredValue },
  });
  await act(async () => {
    fireEvent.click(getByTestId('button'));
  });
});

test('submit value more than max', async () => {
  const newEnteredValue = '500';
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      max="200"
      onSubmit={value => {
        expect(value).toBe(Number(200));
      }}
      loading={false}
    />,
  );
  fireEvent.change(getByTestId('input'), {
    target: { value: newEnteredValue },
  });
  await act(async () => {
    fireEvent.click(getByTestId('button'));
  });
});
test('submit value on disabled prop', async () => {
  const { getByTestId } = render(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      disabled
      onSubmit={value => {
        expect(value).toBe(Number(200));
      }}
      loading={false}
    />,
  );
  expect(getByTestId('button')).toBeDisabled();
});

test('default matches snapshot', () => {
  const dom = renderer.create(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
      loading={false}
    />,
  );
  expect(dom).toMatchSnapshot();
});
test('with spinner matches snapshot', () => {
  const dom = renderer.create(
    <InputContainer
      title={titleContent}
      buttonText={buttonContent}
      placeholder=""
      onSubmit={() => {}}
      loading={true}
    />,
  );
  expect(dom).toMatchSnapshot();
});
