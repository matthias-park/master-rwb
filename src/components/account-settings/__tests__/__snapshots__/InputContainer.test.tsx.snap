// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default matches snapshot 1`] = `
<div
  className="input-container mb-4"
>
  <p
    className="input-container__title text-14 mb-2"
    data-testid="title"
  >
    titlePlaceholder
  </p>
  <div
    className="w-100 form-group"
  >
    <input
      className="input-container__input form-control"
      data-testid="input"
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder=""
      type="number"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    data-testid="button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    buttonPlaceholder
  </button>
</div>
`;

exports[`with spinner matches snapshot 1`] = `
<div
  className="input-container mb-4"
>
  <p
    className="input-container__title text-14 mb-2"
    data-testid="title"
  >
    titlePlaceholder
  </p>
  <div
    className="w-100 form-group"
  >
    <input
      className="input-container__input form-control"
      data-testid="input"
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder=""
      type="number"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    data-testid="button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="mr-1 spinner-border spinner-border-sm"
      data-testid="spinner"
      role="status"
    />
    buttonPlaceholder
  </button>
</div>
`;
